### DISCUSSION TITLE
Run example failed in IDEA

### DISCUSSION THREAD

**Discussion by Roc-00** [open-telemetry/opentelemetry-java#3374] [Q&A]
![image](https://user-images.githubusercontent.com/84842773/124595729-f6605800-de93-11eb-8d8e-9f0178725746.png)
why?

**Comment by jkwatson** [comment]
Can you provide a little bit more detail about what you did you make this happen?

**Reply by Roc-00** [reply]
I don't know, I just downloaded and opened it with IDEA.
![image](https://user-images.githubusercontent.com/84842773/125012452-cfc13d80-e09c-11eb-8f33-1678ac2ed31b.png)

**Reply by jkwatson** [reply]
How did you "download it"? Did you try cloning the repository and running `./gradlew build`?

**Reply by Roc-00** [reply]
Yes, I cloned the repository.And now I know why I failed. I will download JDK 11 and try again.Thank you!
![image](https://user-images.githubusercontent.com/84842773/125016484-03539600-e0a4-11eb-8df5-2eed92d60351.png)
![image](https://user-images.githubusercontent.com/84842773/125016508-0d759480-e0a4-11eb-91f1-602702171ccc.png)

**Reply by jkwatson** [reply]
aha! Yes, you need java 11 to run the build. 👍🏼

**Reply by Roc-00** [reply]
after I updated my jdk
![image](https://user-images.githubusercontent.com/84842773/125285324-1f2d9500-e34d-11eb-8de4-10da120cd739.png)

**Reply by jkwatson** [reply]
can you provide some more details about your environment and what gradle command line you're running? Also, if you run with `--info` do you get any more useful information?

**Reply by Roc-00** [reply]
It was a network problem, and I am OK now. Thank you for you answer！

**Reply by Roc-00** [reply]
I still have a question about why metrics API is almost full of interfaces. Is metrics in the SDK the implementation class of metrics in the API? I'm trying to use open telemetry metrics to collect metrics, but since I have no experience at all, can you teach me how to use the API for the first time? Give me an example. Or is there a document? Or do I need to implement the API interface? Here is my idea, please see if it is correct.
![image](https://user-images.githubusercontent.com/84842773/125881296-8eff60a2-9dfd-412b-86b9-8f81db521c62.png)

**Reply by jkwatson** [reply]
take a look at the examples directory in the repository. There are metrics examples in there.

**Reply by jkwatson** [reply]
But, be aware that these APIs are *not* stable. we are actively in the process of re-working them and they will change significantly in the coming months.

**Reply by Roc-00** [reply]
OK, I will pay close attention to it. Thank you for your reply

**Reply by Roc-00** [reply]
I can not understand the usage of the "observe", I can only see a horizontal line with no change after I run the example of Prometheus.In my imagination, there should be a line changing with the value of incomingmessagecount, but why is the final result a straight line? Or do I need to use the updowncounter？I sincerely hope you can answer my doubts.
![image](https://user-images.githubusercontent.com/84842773/126741504-24d8db6a-239b-443a-ba35-bf8cd00d3c8e.png)

**Reply by jkwatson** [reply]
I think our current instructions in the README don't work on the mac (host networking doesn't really work on the mac). I switched the scraper url in the prometheus.yml to point at `docker.for.mac.localhost:19090` and switched the docker command to expose port 9090, rather than trying to use host networking, and this example worked flawlessly for me.

**Reply by Roc-00** [reply]
I know why I get only one straight line. The cycle output is only 10s, but the acquisition interval is 15s, so only one data can be collected. I suggest lengthening the cycle in example
![image](https://user-images.githubusercontent.com/84842773/126855540-3275e2d8-eae7-4909-ba4a-ef230d71d59d.png)
![image](https://user-images.githubusercontent.com/84842773/126855566-fb9fcb0f-504d-4aec-88b9-1868fea4a22b.png)

**Reply by jkwatson** [reply]
I agree. I usually crank it higher when I verify it's working.

**Reply by Roc-00** [reply]
If I want to export the following data using Prometheus, which instrument is the best?
![image](https://user-images.githubusercontent.com/84842773/126860682-ec929789-3525-4b00-a14f-71707ae8a8b9.png)

**Reply by jkwatson** [reply]
Would you mind opening a new discussion for this, so others will be able to find it?

**Reply by Roc-00** [reply]
Of course not, I will open a new discussion. You can close this.
